controllers/kustomization_gc.go

```diff
@@ -138,7 +138,8 @@ func (kgc *KustomizeGarbageCollector) Prune(timeout time.Duration, name string,
 }
 
 func (kgc *KustomizeGarbageCollector) isStale(obj unstructured.Unstructured) bool {
-	itemChecksum := obj.GetLabels()[fmt.Sprintf("%s/checksum", kustomizev1.GroupVersion.Group)]
```

Comment:
We can't drop the label check in this version. If the label exists but the annotation doesn't, then we have to mark the object as stale otherwise after upgrading to this version the controller will fail to prune the objects reconciled with the older version. If both the label and annotation exists, then we should use the annotation.